### DISCUSSION TITLE
unit testing of NFT ownership in move

### DISCUSSION THREAD

**Discussion by sweetim** [aptos-labs/aptos-developer-discussions#166] [Questions] [answered]
### Describe your question in detail.

i have question about unit testing, i want to test about mint, and make sure the user obtain the NFT object, may i know how can i test it?

since i dont have access to the NFT object
my code is below, how can check if user_1 has the NFT?
https://github.com/sweetim/aptos-counter/blob/master/contract/sources/nft.move#L105

![image](https://github.com/aptos-labs/aptos-developer-discussions/assets/6851767/f058ad4e-86fd-479f-817a-98e936fed2d4)

### What have you tried or looked at?  Or how can we reproduce the error?

i read this guide https://aptos.dev/move/book/unit-testing/, but not much technique on how to test with the new aptos object model

### Which operating system are you using?

Linux (Ubuntu, Fedora, Windows WSL, etc.)

### Which SDK or tool are you using? (if any)

Move aptos std

**Comment by kshitijc1506** [comment, marked as answer]
You can call the `is_owner` or `owner` [function](https://github.com/aptos-labs/aptos-core/blob/4eae1fa3cf34218dd8b1f2bc1572bd043c3e58cd/aptos-move/framework/aptos-framework/sources/object.move#L616) in the object.move module. 

Here's an [example](https://github.com/aptos-labs/aptos-core/blob/4eae1fa3cf34218dd8b1f2bc1572bd043c3e58cd/aptos-move/framework/aptos-token-objects/sources/aptos_token.move#L687) unit test.

**Reply by sweetim** [reply]
the issue here is i dont have access to the object, how can i test it?

**Reply by kshitijc1506** [reply]
**For DA / your code**
In your example, just return the object in your mint function (the nft variable). Use that for testing. 

**Alternative approach (more generalized)**
You can use a named object, retrieve the object address using the seed, and then use [address_to_object](https://github.com/aptos-labs/aptos-core/blob/4eae1fa3cf34218dd8b1f2bc1572bd043c3e58cd/aptos-move/framework/aptos-framework/sources/object.move#L181) to gain access to the object.

**Reply by sweetim** [reply]
for the alternative approach, how can i convert the object to named object? since i have no access to the aptos_stdlib to create that object?

**Reply by kshitijc1506** [reply]
You can just call `object::create_named_object`. Here's the [function](https://github.com/aptos-labs/aptos-core/blob/4eae1fa3cf34218dd8b1f2bc1572bd043c3e58cd/aptos-move/framework/aptos-framework/sources/object.move#L241) in the object.move module. Here's an [example](https://github.com/aptos-labs/aptos-core/blob/94c4f1ec39ef1f9d688ff892651e2cba36f021e2/aptos-move/move-examples/fungible_asset/stablecoin/sources/usdk.move#L99) of how to use it.
